bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -394,46 +405,37 @@ PreProcessor::PreProcessor(shared_ptr<MsgsCommunicator> &msgsCommunicator,
       clientBatchingEnabled_(myReplica.getReplicaConfig().clientBatchingEnabled),
       threadPool_("PreProcessor::threadPool"),
       timers_{timers},
-      // TODO - this is a temporary in-place configuration - should be replaced with a future task to take it from
-      // replica config
-      subpoolsConfig_{
-          {1 << 16, 0, 1400},  // 64KB
-          {1 << 17, 0, 1400},  // 128KB
-          {1 << 18, 0, 1400},  // 256KB
-          {1 << 19, 0, 1400},  // 512KB
-          {1 << 20, 0, 1400},  // 1MB
-          {1 << 21, 0, 1400},  // 2MB
-          {1 << 22, 0, 700},   // 4MB
-          {1 << 23, 0, 700},   // 8MB
-          {1 << 24, 0, 80},    // 16MB
-          {1 << 25, 0, 80},    // 32MB
-      },
-      // TODO - this is a temporary in-place configuration - should be replaced with a future task to take it from
+      dynamicSubpoolsConfig_{YAML::Load(myReplica_.getReplicaConfig().get<std::string>(
```

Comment:
i dont think we need to initialize this member or have it as member at all.
define a non member function above this one called
`static const concordUtil::MultiSizeBufferPool::SubpoolsConfig calcSubpoolsConfig()`

please read the dynamicSubpoolsConfig_ there, decode it and return the config from replica config or default config.

Reply:
Done